### DISCUSSION TITLE
Typescript support

### DISCUSSION THREAD

**Discussion by bastiengrignon** [xavierbriole/react-cookienotice#62] [Ideas]
Discussions about Typescript implementation using functional components

**Comment by xavierbriole** [comment]
The code is inside `next` branch. If you want to check !

**Reply by bastiengrignon** [reply]
I don't know why but I don't have access to the `next` branch, I see it in your repo but it doesn't appear on mine ! 

I don't know if I missed something or if I don't have access to it.

**Reply by xavierbriole** [reply]
You can't see the commit or you can't push `next` branch ?

**Reply by bastiengrignon** [reply]
I made a fork of your repo to be able to make pull requests but on my forked repo I don't have the `next` branch

**Reply by xavierbriole** [reply]
Oh okay, even with a `git fetch` ?

**Reply by bastiengrignon** [reply]
I've figure it out, I thought that Github were doing the right things for me but I needed to do some git command in local !

**Reply by xavierbriole** [reply]
Great :)

**Comment by bastiengrignon** [comment]
So in your Typescript branch you still want ot use react 16 or you need to upgrade it to react 17 too ?

**Reply by xavierbriole** [reply]
We will use React 17 for sure, it's just because the project was started before your pull request ahah

**Reply by bastiengrignon** [reply]
Ok great that's what I was thinking 🙂

**Comment by bastiengrignon** [comment]
I don't know what are you usually doing but when I declare an **interface** for properties of a component, I name them with the name of the component as a suffix
example : component `Buttons` will have this interface `ButtonsProps`

Just a suggestion 😉

**Reply by xavierbriole** [reply]
This is not what I'm usually doing but I like it 😀
The code style still need to be improved, there's no rule but if we can have some conventions, let's do it.
